libcontainer/standard_init_linux.go

```diff
@@ -79,6 +79,14 @@ func (l *linuxStandardInit) Init() error {
 	}
 
 	label.Init()
+
+	// Use Unshare() to enter a new cgroup_namespace before InitializeMountNamespace()
+	if l.config.Config.Namespaces.Contains(configs.NEWCGROUP) && l.config.Config.Namespaces.PathOf(configs.NEWCGROUP) == "" {
+		if err := syscall.Unshare(configs.Syscall_CLONE_NEWCGROUP); err != nil {
```

Comment:
This is seriously not safe. Do not **ever** call `Unshare` from inside Go code. The `Unshare` interfaces do not work in a multithreaded process (and Go programs are multithreaded). This is why `nsexec.c` exists in the first place.